include/EntityExposedFaceSorter.h

```diff
@@ -0,0 +1,50 @@
+/*------------------------------------------------------------------------*/
+/*  Copyright 2014 Sandia Corporation.                                    */
+/*  This software is released under the license detailed                  */
+/*  in the file, LICENSE, which is located in the top-level Nalu          */
+/*  directory structure                                                   */
+/*------------------------------------------------------------------------*/
+
+
+#ifndef EntityExposedFaceSorter_h
+#define EntityExposedFaceSorter_h
+
+#include <stk_mesh/base/BulkData.hpp>
+
+namespace sierra {
+namespace nalu {
+
+//=============================================================================
+// Class Definition
+//=============================================================================
+// EntityExposedFaceSorter
+//=============================================================================
+/**
+ * * @par Description:
+ * - Class that sorts by exposed face ordinal.
+ *
+ * @par Design Considerations:
+ * - We want to sort based on exposed face ordinal - all exposed faces
+ */
+//=============================================================================
+
+class EntityExposedFaceSorter : public stk::mesh::EntitySorterBase {
+
+  public:
+    
+  virtual void sort(stk::mesh::BulkData &bulk, stk::mesh::EntityVector& entityVector) const
+  {
+    stk::mesh::EntityRank entityVecRank = bulk.entity_rank(entityVector[0]);
+
+    if ( entityVecRank == stk::topology::FACE_RANK || entityVecRank == stk::topology::EDGE_RANK ) {
+      std::sort(entityVector.begin(), entityVector.end(),
```

Comment:
Do you want to apply this sort to edges? Or did you just include EDGE_RANK for 2-D meshes where edges are sides? To avoid sorting edges in a 3-D mesh, it might be better to use 'if (entityRank == bulk.mesh_meta_data().side_rank())'

Reply:
Yes, I added EDGE_RANK for the 2D use case. However, I can use the side_rank() just as easily. In general, this will not be active when we have edges in 3D - at least at present. I will make the change now.

Reply:
@alanw0, okay, I made the change. Feel free to squash these commits.